dotcom-rendering/src/web/lib/hasAds.ts

```diff
@@ -0,0 +1,14 @@
+import type { DCRFrontType } from '../../types/front';
+import type { FEArticleType } from '../../types/frontend';
+
+export const hasAds = (pageData: FEArticleType | DCRFrontType): boolean => {
+	let render =
+		!pageData.isAdFreeUser &&
+		pageData.config.abTests.poorDeviceConnectivityVariant !== 'variant';
+
+	if ('shouldHideAds' in pageData && pageData.shouldHideAds) {
+		render = false;
+	}
+
+	return render;
```

Comment:
Could we do early returns here (hope I've not missed a condition) ?

```suggestion
	if (pageData.shouldHideAds || pageData.isAdFreeUser) {
		return false;
	}

	if (pageData.config.abTests.poorDeviceConnectivityVariant === 'variant') {
	  return false;
  }

	return true;
```